docs/tutorials/portable-consent.md

```diff
@@ -0,0 +1,137 @@
+    
+- For a preexisting conversation with one or more messages from a contact that hasn’t had its consent state updated on the network (`unknown`) and your user has taken the action to send a message in response, update the contact consent state to `allowed`. You can consider the response from your user as consenting to the conversation with the contact.
+    
+    For example, you can consider moving the conversation with the contact from the **Requests** tab to the **Messages** tab.
+    
+- For a contact that your user has taken the action to allow, update the contact consent state to `allowed`.
+
+    For example, you can consider displaying the conversation with the allowed contact on the **Messages** tab in your user’s inbox.
+
+**Blocked**
+
+- For a contact that your user has taken the action to block, update the contact consent state to `blocked`.
+    
+    For example, you can consider removing a conversation with the blocked contact from your user’s inbox completely.
+    
+## Build portable consent
+
+Use the following methods to build portable consent in your app.
+
+### Block or allow contacts
+
+To enable your user to block or allow a contact, call the following methods. Note that these functions accept lists, so you can do batch requests.
+
+```tsx
+client.contacts.allow([wantedConvo.peerAddress, wantedConvo.peerAddress]);
+client.contacts.block([spamConvo.peerAddress, unwantedConvo.peerAddress]);
+```
+
+### Refresh the contact consent list
+
+To ensure that you’re using the latest contact consent state, make sure to refresh the consent list from the network. Perform the refresh just in case the consent state has changed on a different device, for example.
+
+```tsx
+client.contacts.refreshConsentList();
+```
+
+### Check if a contact is blocked or allowed
+
+Call the following methods to check if a contact is blocked or allowed.
+
+```tsx
+client.contacts.isAllowed(wantedConvo.peerAddress);
+client.contacts.isBlocked(spamConvo.peerAddress);
+```
+
+### Get a conversation’s consent state
+
+When loading a list of conversations, take into account its current consent state. You can get the `consentState` directly from the conversation.
+
+For example, you might want to display only allowed conversations (conversations with allowed contacts) on a **Messages** tab and “unknown” conversations (conversations with unknown contacts) on a **Requests** tab. For blocked conversations (conversations with blocked contacts), you might want to not display them at all.
+
+```tsx
+const state = await conversation.consentState();
+if (state === "blocked) {
+	// hide the conversation
+}
+```
+
+## Synchronize portable consent
+
+This diagram illustrates the logic for how contact consent states are set in an app and the consent list on the XMTP network. 
+
+Your app must not overwrite a contact consent state in the consent list on the network except if your user performs one of the following explicit actions:
```

Comment:
Just making sure I've worded these correctly and that they cover the select list of actions that should update the consent list.